docs(docs-infra): add npm link documentation for library development

aparzi opened the pull request:
Add comprehensive guide for using npm link with Angular libraries, including required angular.json.

Fixes #61114

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:

Issue Number: #61114

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

aparzi commented:
Hi @JeanMeche @alan-agius4 
any news for this PR?

alan-agius4 commented on adev/src/content/tools/libraries/creating-libraries.md:
```diff
@@ -240,6 +240,106 @@ This means that the TypeScript source can result in different JavaScript code in
+    "cache": {
+      "environment": "local"
+    }
+  }
```
```suggestion
```

alan-agius4 commented on adev/src/content/tools/libraries/creating-libraries.md:
````diff
@@ -240,6 +240,106 @@ This means that the TypeScript source can result in different JavaScript code in
+   npm run build
+   # Or use watch mode for automatic rebuilds
+   npm run build -- --watch
+```
````
This is explained in the links above.

```suggestion

```

alan-agius4 commented on adev/src/content/tools/libraries/creating-libraries.md:
```diff
@@ -240,6 +240,106 @@ This means that the TypeScript source can result in different JavaScript code in
+                "styles": true,
+                "vendor": true
+              },
+              "prebundle": {
```
The prebundle is an option under the dev-server builder not the build.

alan-agius4 reviewed (changes requested):
Thanks for this

alan-agius4 reviewed (changes requested):
Looks like you deleted the entire`Publishing libraries` section. 

In additional please change the 2nd commit to start with `fixup! ` so that the commit will be fixed up once merged, also please format the files changed using

```
pnpm ng-dev format changed
```

alan-agius4 commented:
Ideally keep the commits separate as it's easier to re-review the changes. Thanks.

alan-agius4 commented on adev/src/content/tools/libraries/creating-libraries.md:
```diff
@@ -240,6 +240,67 @@ This means that the TypeScript source can result in different JavaScript code in
+    "your-app": {
+      "architect": {
+        "build": {
+          "options": {
```
```suggestion
          "builder": "@angular/build:application",
          "options": {
```

alan-agius4 commented on adev/src/content/tools/libraries/creating-libraries.md:
```diff
@@ -240,6 +240,67 @@ This means that the TypeScript source can result in different JavaScript code in
+                ]
+              }
+            }
+          }
```
```suggestion
          "builder": "@angular/build:dev-server",
           "options": {
              "prebundle": {
                "exclude": [
                  "my-lib"
                ]
             }
          }
```

Let's not add an entire scope, as the exclude should be either for problematic libraries or just the linked library, definitely not the entire orgs libraries.

alan-agius4 reviewed (changes requested):
Thanks for this, just a couple of other small things.

**Note:** files are still not formatted correctly.

alan-agius4 commented:
> Sorry, I just pushed. I reverted the commits to one because I usually try to have a single commit in PRs.

Please keep separate commits in the future, once merged the commits will be fixed up automatically.

aparzi commented:
> Thanks for this, just a couple of other small things.
> 
> **Note: ** files are still not formatted correctly.

I ran the command but it tells me there are no files to reformat

aparzi committed:
docs(docs-infra): add npm link documentation for library development

Add comprehensive guide for using npm link with Angular libraries, including required angular.json.

fixes #61114

aparzi committed:
fixup! docs(docs-infra): add npm link documentation for library development

alan-agius4 committed:
fixup! docs(docs-infra): add npm link documentation for library development

alan-agius4 reviewed (approved):
LGTM, thanks
